docs/adr/0003-End-to-end-tests.md

```diff
@@ -0,0 +1,57 @@
+# 3. End to end tests
+
+Date: 2023-04-18
+
+## Status
+
+Accepted
+
+## Context
+
+End to end (e2e) tests only run tests against exporters `/metrics` endpoint in OpenShift today (using Prow CI). We want to run them with more cases, like against Grafana, making them more reliable.
+
+## Decision
+
+e2e tests will be more user friendly.
+
+Prow CI will run e2e tests that cover more scenarios and are more reliable.
+
+### How
+
+- We will rewrite the e2e tests script in Python.
+    - easier to write and read it
+    - they will run against Pelorus operator (using test image) and not helm
+
+- We will write better documentation on what e2e tests do (script help and read the docs).
+    - more accessible to new developers of the Pelorus project
+
+- We will log only relevant information on tests run.
+    - Too much log information is not good to identify a simple error
+
+- We will not use `mig-demo-apps` anymore.
+    - make development faster (only one PR and everything in one repository)
+    - do not use yaml files for different tests scenarios
+        - use `image_name` option instead of `source_url` for faster runs (using test image)
+    - use the simplest application possible for tests (faster tests runs)
+
+- New commits/deployments/issues will be created for each test scenario.
```

Comment:
I do not understand this point.

Reply:
I think this is already done for commit/deploy, but issues are always used the old ones

Reply:
the failure exporter tests (maybe committime/deploytime as well) use issues that already exist to run tests ([like this one](https://github.com/konveyor/mig-demo-apps/issues/82)). we should create them during the test run

Reply:
If we create issues during the test run we won't be able to finish test quickly. Our queries and numbers are based on time series, so I think we will be missing the long enough period of time to correctly capture numbers and graphs represented in grafana.

Think of the situation, where the failure is opened and then closed right away by e2e test (remember we don't want our e2e tests run for a long time, so we don't want to wait 5-10minutes between those events). Will this give us enough time window to do any tests against mean time to restore? The minimum Grafana time window is 5 minutes and I think open/close events should happen outside of minimum time window.